source/server/connection_handler_impl.cc

```diff
@@ -185,30 +185,43 @@ void ConnectionHandlerImpl::ActiveListener::onAccept(
 }
 
 void ConnectionHandlerImpl::ActiveListener::newConnection(Network::ConnectionSocketPtr&& socket) {
-  Network::ConnectionPtr new_connection = parent_.dispatcher_.createServerConnection(
-      std::move(socket), config_.transportSocketFactory().createTransportSocket());
+  // Find matching filter chain.
+  const auto filter_chain = config_.filterChainManager().findFilterChain(
+      const_cast<Network::ConnectionSocket*>(socket.get()));
```

Comment:
I think you can lose the const_cast here when you switch to const reference passing? (In general const_cast is an anti-pattern)

Reply:
Done.